OpenSSL/test/test_ssl.py

```diff
@@ -1782,6 +1784,212 @@ def select(conn, options):
 
 
 
+class ApplicationLayerProtoNegotiationTests(TestCase, _LoopbackMixin):
+    """
+    Tests for ALPN in PyOpenSSL.
+    """
+    # Skip tests on versions that don't support ALPN.
+    if _lib.Cryptography_HAS_ALPN:
+
+        def test_alpn_success(self):
+            """
+            Tests that clients and servers that agree on the negotiated ALPN
```

Comment:
Can strike the "Tests that ..." part of this and other test method docstrings if you want.  I think pyOpenSSL is currently inconsistent about this but the most-recently-preferred style assumes the reader can figure out this is a test without the extra hint. :smile: